make-rules/shared-macros.mk

```diff
@@ -26,8 +26,8 @@
 # whenever PATH is to be defined there:
 #     PATH = $(PATH.illumos)
 #     PATH = $(PATH.gnu)
-PATH.illumos =	$(subst $(space),:,$(strip $(PATH.prepend))):$(USRBINDIR$(BITS)):$(USRBINDIR):$(GNUBIN):$(USRSBINDIR$(BITS)):$(USRSBINDIR)
-PATH.gnu =	$(subst $(space),:,$(strip $(PATH.prepend))):$(GNUBIN):$(USRBINDIR$(BITS)):$(USRBINDIR):$(USRSBINDIR$(BITS)):$(USRSBINDIR)
+PATH.illumos =	$(subst $(space),:,$(strip $(PATH.prepend))):$(or $(USRBINDIR$(BITS)),$(USRBINDIR64)):$(USRBINDIR):$(GNUBIN):$(or $(USRSBINDIR$(BITS)),$(USRSBINDIR64)):$(USRSBINDIR):$(HOME)/.cargo/bin
+PATH.gnu =	$(subst $(space),:,$(strip $(PATH.prepend))):$(GNUBIN):$(or $(USRBINDIR$(BITS)),$(USRBINDIR64)):$(USRBINDIR):$(or $(USRSBINDIR$(BITS)),$(USRSBINDIR64)):$(USRSBINDIR):$(HOME)/.cargo/bin
```

Comment:
Please do not add `$(HOME)` to `$(PATH)`.  This will cause several problems:

- it is not guaranteed that `HOME` is set (think of `env - gmake publish`)
- an user could have basically anything in his `$(HOME)/.cargo/bin` and this could have strange side effects to the build framework and cause hard to debug issues.

Thank you.